spec/dummy-app/spec/features/sign_in_spec.rb

```diff
@@ -162,4 +162,39 @@
       end
     end
   end
+
+  context 'older Recognisable sessions are still valid' do
+    let!(:different_user_agent) { 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_4) AppleWebKit/605.1.15 (KHTML, like Gecko)' }
+    let!(:different_accept_header) { 'text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8' }
+    let!(:different_recognisable_session_values) {{
+      recognisable_id: user.id,
+      recognisable_type: 'User',
+      user_agent: different_user_agent,
+      accept_header: different_accept_header
+    }}
+
+    before do
+      # This creates a recognisable session.
+      visit '/'
+      click_link 'Log in'
+      fill_in 'Email', with: user.email
+      fill_in 'Password', with: user.password
+      click_button 'Log in'
+      click_link 'Log out'
+
+      # Create a recognisable session from a different device.
+      FactoryBot.create :recognisable_session, different_recognisable_session_values
+    end
+
+    it 'works and does not send an email' do
```

Comment:
I'm still not 100% sure what this is testing... You're signing in, it creates a session record, then you make a new one with all new details, but it still let's them sign in. Oh wait...is it just testing that the system doesn't only look at the most recent session?

Reply:
Yes. Third time lucky?